### PR TITLE
Added additional OpenStack credentials for DomainID, TenantID, UserDomainName

### PR DESCRIPTION
**Pull request by afritzler**
**What this PR does / why we need it**:
This PR adds additional support for
```
DomainID
TenantID
UserDomainName
```
in the OpenStack machine class secret.

**Which issue(s) this PR fixes**:
This PR is part of https://github.com/gardener/gardener-extensions/issues/154

**Release note**:
```improvement operator
Added additional OpenStack credentials for DomainID, TenantID, UserDomainName
```

### CODE REVIEW

File: pkg/driver/driver_openstack.go [hunk 1]
```diff
@@ -356,16 +282,21 @@ func (d *OpenStackDriver) createNeutronClient() (*gophercloud.ServiceClient, err
...
 	}
+
+	// optional OS_USER_DOMAIN_NAME
+	userDomainName := d.CloudConfig.Data[v1alpha1.OpenStackUserDomainName]
+
 	domainName, ok := d.CloudConfig.Data[v1alpha1.OpenStackDomainName]
-	if !ok {
-		return nil, fmt.Errorf("missing %s in secret", v1alpha1.OpenStackDomainName)
+	domainID, ok2 := d.CloudConfig.Data[v1alpha1.OpenStackDomainID]
+	if !ok && !ok2 {
```
Comment (prashanth26) [inline pkg/driver/driver_openstack.go]: So both of them have to be present right? Just confirming.

File: pkg/driver/driver_openstack.go [hunk 2]
```diff
@@ -356,16 +282,21 @@ func (d *OpenStackDriver) createNeutronClient() (*gophercloud.ServiceClient, err
...
+		return nil, fmt.Errorf("missing %s or %s in secret", v1alpha1.OpenStackDomainName, v1alpha1.OpenStackDomainID)
 	}
 	tenantName, ok := d.CloudConfig.Data[v1alpha1.OpenStackTenantName]
-	if !ok {
-		return nil, fmt.Errorf("missing %s in secret", v1alpha1.OpenStackTenantName)
+	tenantID, ok2 := d.CloudConfig.Data[v1alpha1.OpenStackTenantID]
+	if !ok && !ok2 {
```
Comment (prashanth26) [inline pkg/driver/driver_openstack.go]: Same as above.

File: pkg/driver/driver_openstack.go (hunk 1 above)
Comment (afritzler) [reply pkg/driver/driver_openstack.go]: At least one of them.

**Comment by afritzler** [comment]
Validation would be great. But we only validate the `OpenStackMachineClassSpec` and not the infrastructure secret. Since this change only touches the secret and not the machine spec I currently don't see a point where we could add validation here.

**Comment by prashanth26** [comment]
Okay makes sense. Merging this PR then.

**Pull request merged**